### PR TITLE
feat(react): refactor composed modal to functional component

### LINKED ISSUE #10012: [ComposedModal]: upgrade to functional component

### PR DESCRIPTION
**Pull request by abbeyhrt**
Closes #10012 

Refactors `ComposedModal` to functional component and updates tests. 

#### Testing / Reviewing

Pull down PR and confirm that ComposedModal works as expected

### CODE REVIEW

File: packages/react/src/components/ComposedModal/next/ComposedModal.js [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    containerClassName,
+    danger,
+    onClose,
+    onKeyDown,
+    open,
+    preventCloseOnClickOutside,
+    selectorPrimaryFocus,
+    selectorsFloatingMenus,
+    size,
+    ...other
```
Comment (sstrubberg) [inline packages/react/src/components/ComposedModal/next/ComposedModal.js]:
```suggestion
    ...rest
```

style nit

File: packages/react/src/components/ComposedModal/next/ComposedModal.js [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    }
+
+    if (innerModal.current) {
+      focusButton(innerModal.current);
+    }
+  }, [open, selectorPrimaryFocus]);
+
+  return (
+    <div
+      {...other}
```
Comment (sstrubberg) [inline packages/react/src/components/ComposedModal/next/ComposedModal.js]:
```suggestion
      {...rest}
```

style nit

**Review by sstrubberg — CHANGES_REQUESTED** [review]
This is great! Couple style nits and I noticed that when I use the keyboard and tab my way over to the close button and hit "enter", it doesn't close the modal. It will, however, close it in the current version of ComposedModal.

**Review by dakahn — APPROVED** [review]
nice! great job on the testing as well. approved, pending scotts changes 👍🏾

**Comment by abbeyhrt** [comment]
@sstrubberg which story are you looking at? I am able to tab to the Close icon and use it but I might be looking at the wrong story! Separately, that might be because of changes coming from the `ModalHeader` refactor instead of this one, since the close icon button and functionality is coming from that component.

**Comment by sstrubberg** [comment]
> @sstrubberg which story are you looking at? I am able to tab to the Close icon and use it but I might be looking at the wrong story! Separately, that might be because of changes coming from the `ModalHeader` refactor instead of this one, since the close icon button and functionality is coming from that component.

Super weird. I can't recreate it. Ah well. Nicely done!

**Review by sstrubberg — APPROVED** [review]

**Pull request merged**